packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,444 @@
+  ///
+  /// The [child] argument must not be null.
+  const WidgetInspector({Key key, this.child}) : super(key: key);
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  @override
+  _WidgetInspectorState createState() => new _WidgetInspectorState();
+}
+
+class _WidgetInspectorState extends State<WidgetInspector>
+    with WidgetsBindingObserver {
+  _WidgetClient _client;
+
+  Offset _lastPointerLocation;
+
+  /// Selected [RenderObject] being inspected.
+  ///
+  /// Only RenderObjects that are RenderBox instances are currently supported.
+  RenderBox _selected;
+
+  /// Whether the inspector is in select mode.
+  ///
+  /// In select mode pointer interactions trigger widget selection instead of
+  /// normal interactions.
+  /// Not in select mode, the selected widget is highlighted but the application
+  /// can be interacted with normally.
+  bool isSelectMode = false;
+
+  /// Key tracking where the root in the widget tree
+  final GlobalKey _ignorePointerKey = new GlobalKey();
+
+  @override
+  void initState() {
+    super.initState();
+    // TODO(jacobr): We shouldn't reach out to the WidgetsBinding.instance
+    // static here because we might not be in a tree that's attached to that
+    // binding. Instead, we should find a way to get to the PipelineOwner from
+    // the BuildContext.
+    _client = new _WidgetClient(WidgetsBinding.instance.pipelineOwner)
+      ..addListener(_update);
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    _client
+      ..removeListener(_update)
+      ..dispose();
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeMetrics() {
+    setState(() {
+      // The root transform may have changed, we have to repaint.
+    });
+  }
+
+  void _update() {
+    SchedulerBinding.instance.addPostFrameCallback((Duration timeStamp) {
+      // We want the update to take effect next frame, so to make that
+      // explicit we call setState() in a post-frame callback.
+      if (mounted) {
+        // If we got disposed this frame, we will still get an update,
+        // because the inactive list is flushed after the widget updates
+        // are transmitted to the widget clients.
+        setState(() {
+          // The generation of the _WidgetInspectorListener has changed.
+        });
+      }
+    });
+  }
+
+  void _inspectAt(Offset position) {
+    if (!isSelectMode)
+      return;
+
+    final HitTestResult result = new HitTestResult();
+    final RenderIgnorePointer ignorePointer =
+        _ignorePointerKey.currentContext.findRenderObject();
+    final RenderBox userRender = ignorePointer.child;
+    // TODO
+    userRender.hitTest(result, position: position);
+
+    setState(() {
+      if (result.path.isNotEmpty) {
+        _selected = result.path
+            .firstWhere(((HitTestEntry t) => t.target is RenderBox))
```

Comment:
Well, if you're trying to debug a sliver, then not showing slivers will be a problem. If you're trying to debug a sector based layout, not showing sectors will be a problem. But I don't know how you'd represent those widgets when painting, either. Sectors in particular aren't rectangles, for example. Slivers are pretty complicated beasts too, with multiple relevant rectangular geometries (paintExtent, layoutExtent, hitTestExtent, etc).